drivers/periph_common/init_leds.c

```diff
@@ -35,28 +35,32 @@ void led_init(void)
         return;
     }
 
-#ifdef LED0_PIN
+    /* The condition is dual: We don't init if the LED is absent (eg. when a
+     * LEDn_PIN is defined, but there is a higher level driver such as SAUL PWM that makes the
+     * direct use impossible), but we also don't init if there is no pin (eg.
+     * on native where there is a different mechanism for LEDs). */
+#if defined(LED0_IS_PRESENT) && defined(LED0_PIN)
```

Comment:
I'm not very happy with this special handling needed for `native`, but can't think of a better way either :/

Reply:
I don't even think it's special handling. If any other platform had an LED that did not go though the GPIO subsystem, it would also not have an LEDn_PIN. Granted, I can't think of why that would happen (even if someone were to build an uncased microcontroller with an LED on the same die, and that'd be weird already for various reasons, chances are that'd be connected via the GPIO subsystem).

Reply:
Alright, that's a way of putting it that makes me feel slightly better about it :)